flowauth/frontend/src/ServerAdminDetails.js

```diff
@@ -114,9 +130,66 @@ class ServerAdminDetails extends React.Component {
 
   handleTextChange = name => event => {
     this.setState({
-      [name]: event.target.value
+      pageError: false,
+      errors: ""
     });
-  };
+    var state = {
+      [name]: event.target.value
+    };
+    if (name === "name") {
+      var letters = /^[A-Za-z0-9_]+$/;
+      let servername = event.target.value;
+      if (servername.match(letters) && servername.length <= 120) {
+        state = Object.assign(state, {
+          name_helper_text: ""
+        });
+      } else if (servername.length == 0) {
+        state = Object.assign(state, {
+          name_helper_text: "Server name can not be blank."
+        });
+      } else if (!servername.match(letters)) {
+        state = Object.assign(state, {
+          name_helper_text: "Server name may only contain letters, numbers and underscores."
+        });
+      } else {
```

Comment:
Very minor comment: I personally like to be explicit about the condition that triggers a conditional branch. In this case it's obvious that the `else` branch can only be triggered if the servername is too long, but it's easy to later add another condition in the `if` statement and forget to update the `else` branch, and then the error message displayed is inconsistent with the condition that triggered it.

How about turning this `else` branch into another `else if`?
```suggestion
      } else if (servername.length > 120) {
```
Of course then we need to add a catch-all else branch which blows up if it's triggered (because it should never be reached). I'm not sure what a good way to "blow up" is in Javascript land, though, so maybe this isn't a feasible approach. ;)